common/djangoapps/course_modes/models.py

```diff
@@ -30,6 +30,9 @@ class CourseMode(models.Model):
     We would like to offer a course in a variety of modes.
 
     """
+    class Meta:
+        app_label = "course_modes"
+
```

Comment:
If you look at the deprecation warning, it says: `"or else was imported before its application was loaded."` That's what is happening here - `course_modes.models` is being imported before the `course_modes` app is loaded. The stack looked like this:
```
  /edx/app/edxapp/edx-platform/manage.py(112)<module>()
    110
    111     startup = importlib.import_module(edx_args.startup)
--> 112     startup.run()
    113
    114     from django.core.management import execute_from_command_line

  /edx/app/edxapp/edx-platform/cms/startup.py(31)run()
     29         enable_comprehensive_theme(settings.COMPREHENSIVE_THEME_DIR)
     30
---> 31     django.setup()
     32
     33     autostartup()

  /edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/__init__.py(18)setup()
     14     from django.conf import settings
     15     from django.utils.log import configure_logging
     16
     17     configure_logging(settings.LOGGING_CONFIG, settings.LOGGING)
---> 18     apps.populate(settings.INSTALLED_APPS)

  /edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/apps/registry.py(85)populate()
     83                     app_config = entry
     84                 else:
---> 85                     app_config = AppConfig.create(entry)
     86                 if app_config.label in self.app_configs:
     87                     raise ImproperlyConfigured(

  /edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/apps/config.py(86)create()
     84             # which may specify an app config class with default_app_config.
     85             # Otherwise, entry is a path to an app config class or an error.
---> 86             module = import_module(entry)
     87
     88         except ImportError:

  /usr/lib/python2.7/importlib/__init__.py(37)import_module()
     34                 break
     35             level += 1
     36         name = _resolve_name(name[level:], package, level)
---> 37     __import__(name)
     38     return sys.modules[name]

  /edx/app/edxapp/edx-platform/common/djangoapps/static_replace/__init__.py(9)<module>()
      7
      8 from static_replace.models import AssetBaseUrlConfig, AssetExcludedExtensionsConfig
----> 9 from xmodule.modulestore.django import modulestore
     10 from xmodule.modulestore import ModuleStoreEnum
     11 from xmodule.contentstore.content import StaticContent

  /edx/app/edxapp/edx-platform/common/lib/xmodule/xmodule/modulestore/django.py(42)<module>()
     40 # We also may not always have the current request user (crum) module available
     41 try:
---> 42     from xblock_django.user_service import DjangoXBlockUserService
     43     from crum import get_current_user
     44

  /edx/app/edxapp/edx-platform/common/djangoapps/xblock_django/user_service.py(7)<module>()
      5 from opaque_keys.edx.keys import CourseKey
      6 from xblock.reference.user_service import XBlockUser, UserService
----> 7 from student.models import anonymous_id_for_user, get_user_by_username_or_email
      8
      9 ATTR_KEY_IS_AUTHENTICATED = 'edx-platform.is_authenticated'

  /edx/app/edxapp/edx-platform/common/djangoapps/student/models.py(48)<module>()
     46 from xmodule_django.models import CourseKeyField, NoneToEmptyManager
     47
---> 48 from certificates.models import GeneratedCertificate
     49 from course_modes.models import CourseMode
     50 from enrollment.api import _default_course_mode

  /edx/app/edxapp/edx-platform/lms/djangoapps/certificates/models.py(68)<module>()
     66
     67 from config_models.models import ConfigurationModel
---> 68 from course_modes.models import CourseMode
     69 from instructor_task.models import InstructorTask
     70 from util.milestones_helpers import fulfill_course_milestone, is_prerequisite_courses_enabled

> /edx/app/edxapp/edx-platform/common/djangoapps/course_modes/models.py(28)<module>()
     26
     27 import ipdb;ipdb.set_trace()
---> 28 class CourseMode(models.Model):
     29     """
     30     We would like to offer a course in a variety of modes.
```

I fixed the dependency by pushing the `certificates` import of `course_model` down into the methods that need the imports.  Is there a better fix here? Totally open to something else.